### DISCUSSION TITLE
Request to add XChaCha20-SIV and signcryption to libsodium?

### DISCUSSION THREAD

**Discussion by michelp** [jedisct1/libsodium#1168] [Ideas]
Hello,

I have gotten great use out of both [XChaCha20-SIV](https://github.com/jedisct1/libsodium-xchacha20-siv) and [signcryption](https://github.com/jedisct1/libsodium-signcryption) mini-libraries such that I include them verbatim into [pgsodium](https://github.com/michelp/pgsodium) for use from SQL.  It seems to me that these functions are quite useful and well tested, and that they would be useful to others if they were included in libsodium, although I notice it wasn't mentioned in #1165 so maybe I'm wrong about that.  Any thoughts?  Thanks!

Update: this may also apply to #1151 " have a use-case for users needing both a signing key and an encryption key"

**Comment by jedisct1** [comment]
Well... if we have more than one user, certainly :)

**Comment by samuel-lucas6** [comment]
+1 for XChaCha20-SIV since it's also committing. I would prefer Encrypt-then-MAC for my use cases, but I'm guessing that's a bit much given the number of AEADs already present, especially with AEGIS coming.
